pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -112,6 +114,12 @@ init_remote(){
     local url=$1
     clean_git init --initial-branch=master
     clean_git remote add origin "$url"
+    if [ -n "$sparseCheckout" ]
+    then
+        git config remote.origin.partialclonefilter "blob:none"
+        git sparse-checkout init
+        echo "$sparseCheckout" > .git/info/sparse-checkout
```

Comment:
Did you consider using `git sparse-checkout set --stdin`?

Reply:
Done.